src/projectscene/internal/projectsceneconfiguration.cpp

```diff
@@ -17,6 +17,17 @@ static const muse::Settings::Key MOUSE_ZOOM_PRECISION(moduleName, "projectscene/
 static const muse::Settings::Key INSERT_SILENCE_DURATION(moduleName, "projectscene/insertSilenceDuration");
 static const muse::Settings::Key INSERT_SILENCE_DURATION_FORMAT(moduleName, "projectscene/insertSilenceDurationFormat");
 
+ProjectSceneConfiguration::ProjectSceneConfiguration()
+{
+    dispatcher()->reg(this, "toggle-effects", [this] {
```

Comment:
Configurators should not know about the dispatcher, they are intended only for storing information. Imagine that we need to add additional logic to show/hide the panel, for example, access the track list and check something. Then we can't add this logic here, because it will break the architecture, the configurator becomes a service, it should be as simple as possible. 

The dispatcher must be moved to the model and the model must show/hide the panel and save the information to the configurator. Or as in your version, change the visibility in the configurator, and when receiving the notification - emit data change signal.